### PR TITLE
Add components and hosts indexes

### LINKED ISSUE #25: Add component and host indexes
After #21, each PR page will have a component and host tag. Add per-component and per-host index pages so interested users can look for all PRs by a component. I expect the URL scheme will be eg `https://bitcoin-core-review-club.github.io/component/wallet`, `https://bitcoin-core-review-club.github.io/host/jonatack`, etc

### PR DESCRIPTION
**Pull request by jnewbery**
Adds `/meetings-components` and `/meetings-hosts` which list the meetings by component and host, respectively.

Other small changes:

- previous_meetings is renamed to meetings
- the page.component tag is changed to page.components so a meeting can be tagged as more than one component. PR 10823 is restored to `mempool` and `policy` (which is what it was listed under prior to #21

Resolves #25

### CODE REVIEW

File: _includes/linkers/meetings-pages.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% capture /dev/null %}
+{% if page.url == "/meetings/" %}
+  {% assign _index_links = _index_links | append: "<span style='font-weight: bold'>By date</span>" %}
```
Comment (jonatack) [inline _includes/linkers/meetings-pages.md]: These links and stylings aren't working for me. The `page.url` values and the `<span style>` tags both seem to have issues.

File: meetings-hosts.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+layout: default
+title: Previous Meetings
```
[the remark this answers is not in the document]
Comment (jonatack) [reply meetings-hosts.html]: Proposed in #29.

File: meetings-components.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+layout: default
+title: Previous Meetings
```
[the remark this answers is not in the document]
Comment (jonatack) [reply meetings-components.html]: Proposed in #29.

File: meetings.html [hunk 4]
```diff
@@ -2,16 +2,23 @@
 layout: default
 title: Previous Meetings
```
[the remark this answers is not in the document]
Comment (jonatack) [reply meetings.html]: Proposed in #29.

File: _includes/linkers/meetings-pages.md (hunk 1 above)
Comment (jonatack) [reply _includes/linkers/meetings-pages.md]: Proposed fix in #29.

File: meetings-hosts.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+<div class="Home">
+    <h2 class="Home-posts-title">Previous meetings</h2>
+    <p>{% include linkers/meetings-pages.md %}</p>
+    
+    {% for host in hosts %}
+    <h3 id="{{host}}">{{host}}</h3>
+      <ul>
+      {% for post in site.posts %}
+      {% if post.host == host %}
+        <li><div class="Home-posts-post">
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply meetings-hosts.html]: I've added tags, but I prefer not to add the date (it seems unnecessary - users can look at meetings by date if they want the date)

File: meetings-components.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    <h2 class="Home-posts-title">Previous meetings</h2>
+    <p>{% include linkers/meetings-pages.md %}</p>
+    
+    {% for component in components %}
+    <h3 id="{{component}}">{{component}}</h3>
+      <ul>
+      {% for post in site.posts %}
+      {% if post.components contains component %}
+        <li><div class="Home-posts-post">
+            <a class="Home-posts-post-title" href="{{ post.url }}">{{ post.title }}</a>
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply meetings-components.html]: I've added the host, but not the date (it seems unnecessary - users can look at meetings by date if they want the date)

File: meetings.html [hunk 7]
```diff
@@ -2,16 +2,23 @@
...
     {% capture posttime %}{{post.date | date: '%s'}}{% endcapture %}
+    {% capture components %}
+    {%- for comp in post.components -%}
+    <a href="/meetings-components/#{{comp}}">{{comp}}</a>{% unless forloop.last %}, {% endunless %}
+    {%- endfor -%}
+    {% endcapture %}
     {% if posttime < nowunix %}
         <div class="Home-posts-post">
           <span class="Home-posts-post-date">{{ post.date | date_to_string }}</span>
           <span class="Home-posts-post-arrow">&raquo;</span>
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply meetings.html]: done

File: _includes/linkers/meetings-pages.md (hunk 1 above)
Comment (jnewbery) [reply _includes/linkers/meetings-pages.md]: Thanks!

**Comment by jnewbery** [comment]
Thanks for the review @jonatack . I've taken all of your suggestions except changing the titles to "Previous meetings (by date|by Component|by host (in alphabetical order))". I think it looks a bit odd to have a title that says "by xx" and then have the three indexes below:

image

I've taken your commits and added a final commit that addresses your other comments.

One final question. I'm not sure whether 'meeting' is the right word here. Perhaps we should change it to be `Previous PRs`/`Previous PRs by date`/`Previous PRs by component`/ `Previous PRs by host`. Thoughts?

**Comment by jonatack** [comment]
Good call on the titles. The titles idea came to me when the links weren't indicating which page was being viewed. Now that they do indicate it, the long titles were a bit much and confusing.

Meetings versus PRs: On the face of it, PRs seems good and would re-center the site around PRs. But then -- and maybe I'm overthinking it -- it could be somewhat confusing to squash the two together. Meetings and PRs are two separate domains, each with their own attributes. For example, the meaning of a date for meetings and PRs isn't the same, and meetings have hosts and participants, whereas PRs have authors and reviewers.

FWIW it might be an idea to add PR `authors` and `created_at` dates. As well as have the club auto-tweet/toot a selection of varying messages.

Back to the topic, if I were to design a database to model the business domains of the "club app", I'd surely want to have a `meetings` table and Meeting class domain model... idem for `pull_requests`, `people` (hosts, participants, PR authors, PR reviewers), `tags` with a `pull_requests_tags` join table, maybe even sponsor `organisations` for server costs and building a dev fund :grin:

**Comment by jonatack** [comment]
OTOH "PR *reviews*" could replace "meetings" while remaining distinct from PRs... I don't have a strong opinion and empathise about not being sure.

**Comment by jnewbery** [comment]
Thanks Jon. I've taken all your commits from #31 except the final one. I didn't like padding the bottom of the page with lots of blank space. Instead I've added a css rule to highlight the target section when navigating to it. Let me know what you think.

**Comment by jonatack** [comment]
Thanks, John. Tested ACK. Aqua is perhaps a bit electric on this otherwise sober-looking, black and white site. Perhaps try a shade of light grey, say, #ccc, #ddd, or #eee? For people like me who use the Dark Reader extension to convert everything in the browser to dark mode, a shade of gray can work well in dark mode, too. I tested aqua and the greys in both normal and dark mode. In dark mode with aqua, readability suffers a bit.

**Comment by jnewbery** [comment]
Thanks John. Good idea with the color. I've changed it to #ddd. I think that looks better.

**Comment by jonatack** [comment]
Tested ACK 7c9fa3a587e17c9a801267aa5709656fbf1c7ba2.

**Comment by jnewbery** [comment]
Thanks Jon!

**Comment by jnewbery** [comment]
hmmm, I'm getting 404 for https://bitcoin-core-review-club.github.io/meetings-components/ (but it's working when I preview locally)

**Comment by jonatack** [comment]
Same, is there an additional step needed for new pages on production deployment?

**Comment by jonatack** [comment]
I'm seeing it for the meetings-components and meetings-hosts pages, e.g. the new ones.

**Pull request merged**